lua/autorun/client/cl_chip_lister.lua

```diff
@@ -0,0 +1,321 @@
+include( "cfc_chip_lister/shared/sh_chip_lister.lua" )
+
+local ID_WORLD = "[WORLD]"
+local CPUS_FORMAT = "%05d"
+local STR_GLOBAL = "Overall Total CPUs: "
+local STR_TOTAL = "Total: "
+local STR_CPUS = " CPUs: "
+local STR_MICROSECONDS = utf8.char( 181 ) .. "s"
+local STR_TITLE = "-----E2/SF Lister-----"
+local STR_TOGGLE = "(Press " .. string.upper( input.LookupBinding( "+use" ) or "e" ) .. "/use to toggle)"
+local STR_ENABLE = "Press " .. string.upper( input.LookupBinding( "+use" ) or "e" ) .. "/use to turn on"
+local STR_WAITING = "Waiting for next update from the server..."
+local RENDER_TARGET_NAME = "cfc_chiplister_rt"
+local MATERIAL_NAME = "cfc_chiplister_screen"
+local FONT_NAME = "CFC_ChipLister_Font"
+local FONT_SIZE = 30
+local MAX_ELEMENTS = 30
+local SCREEN_SIZE = 1024
+local SCREEN_SIZE_HALF = SCREEN_SIZE / 2
+local TOGGLE_DIST = 1500
+local COLOR_BACKGROUND = Color( 0, 0, 0, 255 )
+local COLOR_DIVIDER = Color( 255, 255, 255, 255 )
+local COLOR_TEXT = Color( 255, 255, 255, 255 )
+local COLOR_TEXT_FADED
+local COLOR_MICROS
+local COLOR_WORLD = Color( 150, 120, 120, 255 )
+local COLOR_FADE_OFFSET = Color( 0, 0, -0.65 )
+local COLOR_FADE_MICROS_OFFSET = Color( 0, 0, -0.25 )
+local CHIP_COLORS = {
+    E2 = Color( 216, 34, 45, 255 ),
+    SF = Color( 55, 100, 252, 255 ),
+}
+
+local rtChipLister = GetRenderTarget( RENDER_TARGET_NAME, SCREEN_SIZE, SCREEN_SIZE )
+local INFO_OFFSET_OWNER = 0
+local INFO_OFFSET_CHIP = 0
+local TOGGLE_DIST_SQR = TOGGLE_DIST ^ 2
+
+local matChipLister = CreateMaterial( MATERIAL_NAME, "UnlitGeneric", {
+    ["$basetexture"] = RENDER_TARGET_NAME,
+    ["$model"] = 1,
+} )
+
+local isValid = IsValid
+local rawset = rawset
+local rawget = rawget
+local _colorToHSV = ColorToHSV
+local _hsvToColor = HSVToColor
+local stringLen = string.len
+local stringSub = string.sub
+local stringFormat = string.format
+local teamGetColor = team.GetColor
+local getClass
+local getTeam
+
+local FONT_DATA = {
+    font = "Roboto Mono",
+    extended = false,
+    size = FONT_SIZE,
+    weight = 500,
+    blursize = 0,
+    scanlines = 0,
+    antialias = true,
+    underline = false,
+    italic = false,
+    strikeout = false,
+    symbol = false,
+    rotary = false,
+    shadow = false,
+    Additive = false,
+    outline = false,
+}
+
+do
+    local entityMeta = FindMetaTable( "Entity" )
+
+    getClass = entityMeta.GetClass
+
+
+    local playerMeta = FindMetaTable( "Player" )
+
+    getTeam = playerMeta.Team
```

Comment:
Might wanna test how much effect this has for performance, from my testing localizing meta table functions gives a neglectable performance increase